proto/internal/temporal/server/api/adminservice/v1/request_response.proto

```diff
@@ -602,3 +603,9 @@ message ForceUnloadTaskQueuePartitionRequest {
 message ForceUnloadTaskQueuePartitionResponse {
   bool was_loaded = 1;
 }
+
+message GetConfigurationsRequest {}
+message GetConfigurationsResponse{
+  string cluster_name = 1;
+  map<string, google.protobuf.Value> config_values = 2;
```

Comment:
```suggestion
repeated HostConfig host_config = 1;
```

And add a `HostConfig` message that will embed structured config via a [struct](https://protobuf.dev/reference/protobuf/google.protobuf/#struct):

```proto
message HostConfig {
  // The host that this configuration was loaded from.
  string hostname = 1;
  // The raw dynamic config loaded with all constraints.
  google.protobuf.Struct raw_dynamic_config = 2;
}
```